stl/src/vector_algorithms.cpp

```diff
@@ -1496,5 +1497,121 @@ __declspec(noalias) size_t
     return __std_count_trivial<_Find_traits_8>(_First, _Last, _Val);
 }
 
+} // extern "C"
+
+namespace {
+    __m128i
+#ifdef _M_X64
+        __vectorcall
+#endif
+        _Bitset_to_string_1_step(const uint16_t _Val, const __m128i _Elem0_v, const __m128i _Elem1_v) {
+        const uint16_t _Valsw = _byteswap_ushort(_Val);
+        const __m128i _Vec1   = _mm_cvtsi32_si128(_Valsw);
+        const __m128i _Vec2   = _mm_unpacklo_epi8(_Vec1, _Vec1);
+        const __m128i _Vec3   = _mm_unpacklo_epi8(_Vec2, _Vec2);
+        const __m128i _Vec4   = _mm_unpacklo_epi8(_Vec3, _Vec3);
+        const __m128i _Msk    = _mm_and_si128(_Vec4, _mm_set1_epi64x(0x0102040810204080));
+        const __m128i _Zeros  = _mm_cmpeq_epi8(_Msk, _mm_setzero_si128());
+        const __m128i _Elem0s = _mm_and_si128(_Zeros, _Elem0_v);
+        const __m128i _Elem1s = _mm_andnot_si128(_Zeros, _Elem1_v);
+        const __m128i _Elems  = _mm_or_si128(_Elem0s, _Elem1s);

```

Comment:
This can be improved with:
 * `_mm_shuffle_epi8` input value populating (replaces byteswap and upacks)
 * `_mm_blendv_epi8` output blending (replaces and/andnot/or)
We don't have discrete SSSE3 / SSE4.1 detection, both can be detected as SSE4.2

I think this can be done in a subsequent PR, so that we step gradually, as usually, but can also be done in this.

Reply:
A followup PR would be fine. (It looks like you altered the code after making this comment, but aren't yet using instructions beyond SSE2, so potential improvements are still possible.)